### ISSUE TITLE
Padding is unavailable on any model / PR guidance

### ISSUE THREAD

**Issue by wahabk** [Project-MONAI/MONAI#5406]
**Is your feature request related to a problem? Please describe.**
First of all thank you for making monai :pray: 

MONAI models have a problem where convolution unpadding is unavailable on *any* of the models I've seen. By this I mean there is no option to make the model output a smaller ROI than the input.

This is crucial for applications in patch based pipelines, where it is difficult to fit an entire image into GPU ram which further hampers batch_size. I usually combine MONAI with [torchio](https://torchio.readthedocs.io/)  which includes a patch based dataloader and aggregator based on niftynet. My models get bad predictions at the edges when I train them unpadded.

**Describe the solution you'd like**

An argument added to ConvNet models where a `padding` variable can be `'same'`, or `'valid'`. whether to pad encoding layers for making the output same size or smaller than input. This is added in init as an argument, I will use `monai/networks/nets/attentionunet.py` as an example:

```python
self.padding = padding
if padding == 'same':
    self.pad = None
elif padding == 'valid':
    self.pad = 0
else:
    raise Exception("Only options available for padding is 'same' or 'valid'")
```

This is then fed to the convolution head

```python
head = ConvBlock(spatial_dims=spatial_dims, in_channels=in_channels, out_channels=channels[0], dropout=dropout, pad=self.pad)
reduce_channels = Convolution(
    spatial_dims=spatial_dims,
    in_channels=channels[0],
    out_channels=out_channels,
    kernel_size=1,
    strides=1,
    padding=0,
    conv_only=True,
)
```
I admire recursion as a solution for concisely building an autoencoder but I personally still struggle with it and I'm not sure if only using the head is a good solution since the rest of the blocks arent unpadded. I've avoided changing the padding inside the recursion but this turned out good for me:

My usual input size is 64 cubed, and this unpadding results in an output size of 60 cubed

**Describe alternatives you've considered**
I can't think of alternative solutions, plus I am unsure of how this could apply to transformer models.

**Additional context**
The original unet paper uses unpadded convolutions.

**Please help me learn pull requests!!**
I'm not asking for this to be done but If you can guide me to a good solution I would LOVE to help contribute to monai, I've been wanting to learn PRs. I will make sure to lint/test etc and follow the guidelines
